uml4net.xmireader/Readers/Packages/PackageReader.cs

```diff
@@ -18,47 +18,70 @@
+    using POCO.StructuredClassifiers;
+    using POCO;
     using uml4net.POCO.CommonStructure;
     using uml4net.POCO.Packages;
 
-    using uml4net.xmi.CommonStructure;
-    using uml4net.xmi.SimpleClassifiers;
-    using uml4net.xmi.StructuredClassifiers;
+    using Readers;
 
     /// <summary>
     /// The purpose of the <see cref="PackageReader"/> is to read an instance of <see cref="IPackage"/>
     /// from the XMI document
     /// </summary>
-    public class PackageReader : XmiElementReader
+    public class PackageReader : XmiCommentedElementReader<IPackage>, IXmiElementReader<IPackage>
     {
         /// <summary>
-        /// The <see cref="ILogger"/> used to log
+        /// The <see cref="IXmiElementReader{T}"/> of <see cref="IComment"/>
+        /// </summary>
+        private readonly IXmiElementReader<IClass> classReader;
+
+        /// <summary>
+        /// The <see cref="IXmiElementReader{T}"/> of <see cref="IEnumeration"/>
+        /// </summary>
+        private readonly IXmiElementReader<IEnumeration> enumerationReader;
+
+        /// <summary>
+        /// The <see cref="IXmiElementReader{T}"/> of <see cref="IPackageImport"/>
+        /// </summary>
+        private readonly IXmiElementReader<IPackageImport> packageImportReader;
+
+        /// <summary>
+        /// The <see cref="IXmiElementReader{T}"/> of <see cref="IPrimitiveType"/>
         /// </summary>
-        private readonly ILogger<PackageReader> logger;
+        private readonly IXmiElementReader<IPrimitiveType> primitiveReader;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="PackageReader"/> class.
         /// </summary>
         /// <param name="cache">
         /// The cache in which each <see cref="IXmiElement"/>> is stored
         /// </param>
-        /// <param name="loggerFactory">
-        /// The (injected) <see cref="ILoggerFactory"/> used to setup logging
+        /// <param name="logger">
+        /// The (injected) <see cref="ILogger{T}"/> used to setup logging
         /// </param>
-        public PackageReader(Dictionary<string, IXmiElement> cache, ILoggerFactory loggerFactory = null)
-            : base(cache, loggerFactory)
+        /// <param name="commentReader">The <see cref="IXmiElementReader{T}"/> of <see cref="IComment"/></param>
+        /// <param name="classReader">The <see cref="IXmiElementReader{T}"/> of <see cref="IClass"/></param>
+        /// <param name="enumerationReader">The <see cref="IXmiElementReader{T}"/> of <see cref="IEnumeration"/></param>
+        /// <param name="packageImportReader">The <see cref="IXmiElementReader{T}"/> of <see cref="IPackageImport"/></param>
+        /// <param name="primitiveReader">The <see cref="IXmiElementReader{T}"/> of <see cref="IPrimitiveType"/></param>
+        public PackageReader(IXmiReaderCache cache, ILogger<PackageReader> logger, IXmiElementReader<IComment> commentReader, IXmiElementReader<IClass> classReader,
```

Comment:
very long constructor, in case we stick with DI, i propose we go for property injection with something like https://autofac.readthedocs.io/en/latest/register/prop-method-injection.html#property-injection